投票用のCog作成

sizumita opened the pull request:
投票用のCog作成

sizumita committed:
プロパティの追加

[Add] 人狼に殺されるプレイヤーのプロパティ
[Add] 占い師に占われるプレイヤーのプロパティ
[Add] 投票によって処刑されるプレイヤーのプロパティ

sizumita committed:
プロパティの追加

[Add] プレイヤーが死んでいるかのプロパティを追加

sizumita committed:
インスタンス追加

[Add] botに経過日のインスタンスを追加

sizumita committed:
add cogs/pagenator.py

[Add] ページネーター作成

sizumita committed:
Embedの仕組みの変更

[Change] Embedの構造

sizumita committed:
修正

[Fix] 誤字修正

sizumita committed:
Cog追加

[Add] 投票用のCogを追加
Cogの読み込み

[Change] Cogの読み込み処理を変更

sizumita committed:
ページネーターの修正

[Fix] channelが取得できないバグ
[Fix] 要素数より多いリアクションを押された時のバグ
[Fix] embedを直接渡していたバグ
[Change] inline=Trueにした

sizumita committed:
flake8の修正

sizumita committed:
flake8の修正

sizumita committed:
Merge branch 'master' into feature/votes

daima3629 reviewed (dismissed):
おそらく特に問題ないと思います👍
お疲れ様です！

sizumita committed:
Merge branch 'feature/votes' into feature/pagenator

sizumita committed:
Merge pull request #10 from DiscordBotPortalJP/feature/pagenator

Feature/pagenator

sizumita committed:
Revert "Feature/pagenator"

sizumita committed:
Merge pull request #13 from DiscordBotPortalJP/revert-10-feature/pagenator

Revert "Feature/pagenator"

1ntegrale9 commented on cogs/vote.py:
```diff
@@ -0,0 +1,26 @@
+    def __init__(self, bot):
+        self.bot = bot
+
+    async def furtune(self, player):
```
typo

```suggestion
    async def fortune(self, player):
```

sizumita replied:
pagenatorの方で変更していましたが、こちらではしていませんでしたね

1ntegrale9 commented on cogs/vote.py:
```diff
@@ -0,0 +1,26 @@
+        """人狼"""
+        pass
+
+    async def player_vote(self, players):
```
多分 Vote クラスと名前分けたいからこうしてる？
シンプルに vote の方が良いと思います

```suggestion
    async def vote(self, players):
```

sizumita replied:
コマンド名にvoteコマンドがあるのでplayer_voteにしました

1ntegrale9 replied:
Cog内の名前空間とVoteクラス内の名前空間は別なので、
そこの競合はあまり意識する必要はないと思います

……と思ったらCogの中に各役職のメソッドありますね。
そもそもこれらのメソッドの役割がよくわからくなったので説明をお願いしたいです 🙏

sizumita replied:
投票の処理は三つありますが、全て別の処理なので、それぞれで関数を分けたということです

sizumita replied:
Cogの名前がVoteで、他にVoteクラスは作っていないと思います

sizumita committed:
Update cogs/vote.py

[Fix] typo

Co-authored-by: 1ntegrale9 <st09d23+github@gm.ibaraki-ct.ac.jp>

1ntegrale9 commented on discordbot.py:
```diff
@@ -41,6 +41,26 @@
+# ゲームの経過日(投票、占い、襲撃先をリセット時に追加される)
+bot.days = 0
+
+# 処刑される人
+bot.voted_player = None
+
+# 占いをされる人
+bot.fortuned_player = None
+
+# 人狼に殺される人
+bot.killed_player = None
```
このあたりって必要なんですっけ？
今のところ集計と実行とメッセージ表示が同時の想定なので不要かなと。

sizumita replied:
同時なのですね、わかりました。
ではVoteCogの中の関数にて、集計、実行（ユーザーを殺すなど）、メッセージ表示をすれば良いですか？

1ntegrale9 replied:
そうですね！

sizumita replied:
了解です、消しておきます。

sizumita replied:
https://github.com/DiscordBotPortalJP/werewolf/pull/11/commits/614d0cf62b2d4ad5eaaf414ab1780cb5ab5c5154
done

1ntegrale9 commented on discordbot.py:
```diff
@@ -41,6 +41,26 @@
+bot.killed_player = None
+
+# cogの読み込み
+extensions = [
+    'cogs.status',
+    'cogs.vote',
+]
```
これ変更するなら players も追加してほしいです

```suggestion
extensions = [
    'cogs.status',
    'cots.players',
    'cogs.vote',
]
```

sizumita committed:
Update discordbot.py

[Add] playersCogの読み込み

Co-authored-by: 1ntegrale9 <st09d23+github@gm.ibaraki-ct.ac.jp>

sizumita committed:
変更

by https://github.com/DiscordBotPortalJP/werewolf/pull/11#discussion_r419986017

1ntegrale9 committed:
Fix: コードの全体が見にくいのでコメントの位置を修正

1ntegrale9 committed:
Merge remote-tracking branch 'origin/master' into feature/votes

1ntegrale9 committed:
Fix: プロパティとメソッドの整理

1ntegrale9 committed:
Fix: cogのloadのバグ修正
